### PR TITLE
Add SPDX header

### PR DESCRIPTION
**Pull request by AndyScherzinger**
and

* GH action to check on future changes
* Add badge to readme file

Many already haad SPDX header, thanks to @susnux - so I just completed his work

### CODE REVIEW

**Review by susnux — APPROVED** [review]

File: lib/Controller/LogController.php [hunk 1]
```diff
@@ -1,23 +1,7 @@
...
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
- * GNU Affero General Public License for more details.
- *
- * You should have received a copy of the GNU Affero General Public License, version 3,
- * along with this program.  If not, see <http://www.gnu.org/licenses/>
- *
+ * SPDX-FileCopyrightText: 2015 ownCloud, Inc.
```
Comment (Antreesy) [inline lib/Controller/LogController.php:3]: Shall we add `* SPDX-FileCopyrightText: 2016-2024 Nextcloud GmbH and Nextcloud contributors` here?

File: lib/Controller/PageController.php [hunk 2]
```diff
@@ -1,22 +1,7 @@
...
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
- * GNU Affero General Public License for more details.
- *
- * You should have received a copy of the GNU Affero General Public License, version 3,
- * along with this program.  If not, see <http://www.gnu.org/licenses/>
- *
+ * SPDX-FileCopyrightText: 2015 ownCloud, Inc.
```
Comment (Antreesy) [inline lib/Controller/PageController.php:3]: and here

File: src/components/LogDetailsModal.vue [hunk 3]
```diff
@@ -1,3 +1,7 @@
+<!--
+	SPDX-FileCopyrightText: 2023 Ferdinand Thiessen <rpm@fthiessen.de>
```
Comment (Antreesy) [inline src/components/LogDetailsModal.vue:2]: Shouldn't it be changed with a next contributor? Don't want to belittle anyone's efforts, but IMO would be better to unify all copyrights pre-emptive

**Review by Antreesy — APPROVED** [review]

File: src/components/LogDetailsModal.vue (hunk 3 above)
Comment (AndyScherzinger) [reply src/components/LogDetailsModal.vue:2]: @susnux what do you think? Fine with you? I can do that then.

Comment (susnux) [reply src/components/LogDetailsModal.vue:2]: Yes fine with me, I think we should use the Authors.md for keeping track of authors so :heavy_check_mark:

Comment (Antreesy) [reply src/components/LogDetailsModal.vue]: Might have seen also a couple of copyrights of Côme and Robin :eyes: 

but if CI will be greeen, I can follow-up this later

Comment (susnux) [reply src/components/LogDetailsModal.vue]: I think this is fine, we can ask them or simply add the generic one additionally to it.

**Pull request merged**